src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -50,90 +76,105 @@ public String getUrlForPath(String wikiPath) {
   }
 
   public String getUrlForWiki() {
-    return getUrlForWiki(Configuration.getWikiName());
+    return getUrlForWiki(Configuration.getWikiName(), false);
   }
 
   public String getUrlForWiki(String wikiName) {
-    String prefix = "wikia".equals(wikiName) ? "www." : "";
-    String suffix;
-
-    if (env.contains("dev") && !env.contains(SANDBOX_MERCURY_DEV)) {
-      String devBoxOwner = env.split("-")[1];
-
-      suffix = "." + devBoxOwner + "." + "wikia-dev.com";
-    } else {
-      suffix = wikiName.endsWith("wikia") ? ".com" : ".wikia.com";
-    }
+    return getUrlForWiki(wikiName, false);
+  }
 
-    return composeUrl(prefix, wikiName, suffix);
+  public String getUrlForWiki(boolean addWWW) {
+    return getUrlForWiki(Configuration.getWikiName(), addWWW);
   }
 
-  public String appendQueryStringToURL(String url, String qs) {
-    String separator = url.contains("?") ? "&" : "?";
 
-    String[] filteredUrl = url.split("#");
-    if (filteredUrl.length > 1) {
-      return filteredUrl[0] + separator + qs + "#" + filteredUrl[1];
-    } else {
-      return url + separator + qs;
+  public String getUrlForWiki(String wikiName, boolean addWWW) {
+    final String wikiaName = getWikiaGlobalName(wikiName);
+    EnvType envType = Configuration.getEnvType(this.env);
+
+    String www = "";
+    if (addWWW) {
+      www = "www.";
     }
-  }
 
-  private String composeUrl(String prefix, String wikiName, String suffix) {
-    String overwrittenWikiName = wikiName;
-    String overwrittenPrefix = prefix;
-    if (env != null) {
-      if ((!env.contains("dev") || env.contains(SANDBOX_MERCURY_DEV)) && !"prod".equals(env)) {
-        overwrittenPrefix = env + "." + prefix;
+    switch (envType) {
+      case DEV: {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested code block into a method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1199)